### ISSUE TITLE
Onedrive api: Getting Internal server Error while using action.createLink for a shared file

### ISSUE THREAD

**Issue by Rocky2908** [OneDrive/onedrive-api-docs#237]
I am trying to create a publicly shared link for onedrive files.

I have got onedrive.readwrite scope from the onedrive account(A).

I tried creating a sharing link for the Account A's file using the following call,

https://api.onedrive.com/v1.0/drive/items/{normal_file_id}/action.createLink
and it worked. I got the web url and the status is 200.

Then I shared another Account B's folder with Account A.

Now,

I tried the same by creating a sharing link for the Account A's shared file of Account B using the following call,

https://api.onedrive.com/v1.0/drive/items/{shared_file_id}/action.createLink
Now, I am getting internal server error with status 500.

I tried both these scenarios in Apigee console. getting the same trouble.

Interestingly when I look into the sharing info for the file in Onedrive. I can see a Sharing link created for that file in the name of Apigee Api console.

![screen shot 2015-10-22 at 1 47 12 pm](https://cloud.githubusercontent.com/assets/10181686/10663439/02f8a544-78d9-11e5-836e-6a7cb6a69d12.png)

But, I am always getting 500 errors while working in API. How to rectify this problem?. Any help is appreciated. Thank you.

**Comment by ginach** [comment]
This is a bug in the API. The error is happening after the link creation succeeds but before the API response is sent.

**Comment by Rocky2908** [comment]
Hi Ginach,
Thank you very much for the response and letting me now it's a API bug. Do you have any other workaround to create a sharing link for the shared files?

**Comment by mflalor** [comment]
Hi,
I'm also getting an Internal Server Error but my case is slightly different: Am getting it on the OneDrive for Business root url - https://procq-my.sharepoint.com/_api/v2.0 and receive the message "host web isn’t a personal site".

Note: I have got my access and refresh tokens and can regenerate from the refresh token. I can upload / create files via the above api but I cannot get a shared link using action.createLink. My scope has been set correctly ( I think ) in the Azure AD portal.

Can you confirm this is a bug?

Cheers, Mark

**Comment by ksheode** [comment]
Hi Mark,

This is a bug in createLink and we have checked-in a fix. The fix is expected to be applied worldwide to Production in the next 2 weeks.

-Ketaki

**Comment by mflalor** [comment]
Thanks Ketaki,

We look forward to the fix.

Cheers, Mark

On 31 October 2015 at 07:08, Ketaki Sheode <notifications@github.com> wrote:

> Hi Mark,
>
> This is a bug in createLink and we have checked-in a fix. The fix is
> expected to be applied worldwide to Production in the next 2 weeks.
>
> -Ketaki
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OneDrive/onedrive-api-docs/issues/237#issuecomment-152639588>
> .
>



-- 
Mark Lalor
(w) procq.com.au <http://www.procq.com.au>
(e) markl@procq.com.au
(t) +61 2 8719 0163

**Comment by mxhero-scm** [comment]
+1 to this ticket, we are facing the same issue.

**Comment by rgregg** [comment]
This particular issue (creating a sharing link for a file) should be resolved for most tenants now.

However, there are still some scenarios where createLink will fail for ODB, including trying to share folders or files that are not owned by the logged in user. If one of these scenarios applies to you, please open a new issue.

**Comment by chstanley** [comment]
> This particular issue (creating a sharing link for a file) should be resolved for most tenants now.However, there are still some scenarios where createLink will fail for ODB, including trying to share folders or files that are not owned by the logged in user. If one of these scenarios applies to you, please open a new issue.
> --

I just tried this with a file and i am also getting the `(500) Internal Server Error` message on my ODB account.  I see that there haven't been any posts for two years on this one.  Can anybody else confirm that this is or is not resolved?
thanks!
Charlton

**Comment by chstanley** [comment]
I think this has been resolved now.  Looks like the documentation was updated back on 09/17.  I went and followed the directions and it works now!  (it is possible that i missed this the first time around, not sure really). [https://docs.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_createlink](url)

From powershell, i was able to use `Invoke-RestMethod` and call a `POST` method and pass a URI of `https://{tenant}-my.sharepoint.com/_api/v2.0/drive/items/{normal_file_id}/createLink` and a json object that had
`{`
`    "type": "view",`
`    "scope": "anonymous"`
`}`
which returned an object that i contained the share link! :smiley: 

Previously, i wasnt passing the json object, which is definitely why it wasn't working for me.  Whether it was documentation, my lack of reading skillz (probably this one), or a code commit by the OD Github contributors, I am not sure.  Either way i am happy that is it resolved! :smiley:  

Thanks all!
